File: tester/harmony/rnoh/src/main/ets/RNOHCorePackage/turboModules/VibrationTurboModule.ts
import type { TurboModuleContext } from '../../RNOH/TurboModule';
import { TurboModule } from '../../RNOH/TurboModule';
import { RNOHLogger } from '../../RNOH/RNOHLogger';

export class VibrationTurboModule extends TurboModule {
  public static readonly NAME = 'Vibration';

  private logger: RNOHLogger;

  constructor(protected ctx: TurboModuleContext) {
    super(ctx);
    this.logger = this.ctx.logger.clone("VibrationTurboModule")
  }

  public vibrate(pattern: number) {
    this.ctx.rnAbility.vibration.vibrate(pattern);
  };

  // Android only
  public vibrateByPattern(pattern: Array<number>, repeat: number) {
    this.logger.warn("Vibration::vibrateByPattern is not supported");
  };

  public cancel() {
    this.ctx.rnAbility.vibration.cancel();
  }
}

File: tester/harmony/react_native_openharmony/src/main/cpp/RNOHCorePackage/TurboModules/VibrationTurboModule.cpp
#include "RNOHCorePackage/TurboModules/VibrationTurboModule.h"

using namespace facebook;

namespace rnoh {
VibrationTurboModule::VibrationTurboModule(
    const ArkTSTurboModule::Context ctx,
    const std::string name)
    : ArkTSTurboModule(ctx, name) {
  methodMap_ = {
      ARK_METHOD_METADATA(vibrate, 1),
      ARK_METHOD_METADATA(cancel, 0),
  };
}
} // namespace rnoh
